stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/entity/ChannelConfigEntity.kt

```diff
@@ -1,8 +1,56 @@
 package io.getstream.chat.android.livedata.entity
 
+import androidx.room.Embedded
 import androidx.room.Entity
+import androidx.room.ForeignKey
 import androidx.room.PrimaryKey
-import io.getstream.chat.android.client.models.Config
+import androidx.room.Relation
+import java.util.Date
 
 @Entity(tableName = "stream_chat_channel_config")
-internal data class ChannelConfigEntity(@PrimaryKey var channelType: String, var config: Config)
+internal data class ChannelConfigInnerEntity(
+    @PrimaryKey
+    val channelType: String,
+    val createdAt: Date? = null,
+    val updatedAt: Date? = null,
+    val isTypingEvents: Boolean = false,
+    val isReadEvents: Boolean = false,
+    val isConnectEvents: Boolean = false,
+    val isSearch: Boolean = false,
+    val isReactionsEnabled: Boolean = false,
+    val isRepliesEnabled: Boolean = false,
+    val isMutes: Boolean = false,
+    val maxMessageLength: Int = Int.MAX_VALUE,
+    val automod: String = "",
+    val infinite: String = "",
+    val name: String = "",
+)
+
+@Entity(
+    tableName = "command_inner_entity",
+    foreignKeys = [
+        ForeignKey(
+            entity = ChannelConfigInnerEntity::class,
+            parentColumns = ["channelType"],
+            childColumns = ["channelType"],
+            onDelete = ForeignKey.CASCADE,
+            onUpdate = ForeignKey.CASCADE
+        )
+    ]
+)
+internal data class CommandInnerEntity(
+    val name: String,
+    val description: String,
+    val args: String,
+    val set: String,
+    val channelType: String,
+) {
+    @PrimaryKey
+    var id: Int = name.hashCode() + description.hashCode() + args.hashCode() + set.hashCode() + channelType.hashCode()
```

Comment:
What's the purpose of using this combined hashcode value here? (As a separate thought, perhaps `hashCode()` itself could be used, which hashes the entire data class.)

Reply:
Yes, you're right. Nice catch